### DISCUSSION TITLE
CDC Manual "huh?"

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#64] [General]
![cobol5ug-wtf.png](https://github.com/kej715/DtCyber/assets/136194499/d50689d6-2adb-4a4a-ba62-cb82dc5b49eb)

**Comment by wschaub** [comment]
COBOL 5 user's guide page 21?  I think what they're trying to say is don't do " this " unless you really mean to have the spaces be part of the literal, because it won't ignore the spaces. The bit on the previous page about nonnumeric literals should be a bit more clearly worded.

**Reply by KC5VDJ** [reply]
Dunno, man, dunno.  Just seems ridiculously redundant.

To see what it is about: https://github.com/KC5VDJ/CDC-Manuals

Does anyone know how to get around the 25 meg upload size limit here at github?  The FTN5 manual is done as well, but...

**Comment by kej715** [comment]
Have you tried using git-lfs <https://git-lfs.com>?

**Reply by KC5VDJ** [reply]
Thanks, sorta new to git.

**Comment by philclaridge** [comment]
And/or lightly compressing the PDF using Adobe’s free tool: https://www.adobe.com/uk/acrobat/online/compress-pdf.html

**Reply by KC5VDJ** [reply]
Actually, that I have been doing, by way of the reduce file size checkbox in save as.  It does better than extraction and reencoding the images with 20% and worse jpeg, tried that and still not as good as the adobe reduce.  I could reduce image sizes and try repacking the files, but less res with lots of jpeg noise, I don't think so.   Just grabbed git-lfs

**Comment by wschaub** [comment]
I've just taken the time to look at your version of the installation guide manual. This is great work!  

I've put a few manuals on my google drive that I think are the most essential to have as searchable manuals for day to day usage/administration of NOS2.8.7

You will find:

-  the most recent system programmer's instant
-  the 1997 edition of volume 3 (system commands)
- the 1996 Analysis handbook. 
-  1988 version of the system programmer's instant
- the diagnostic index. which contains a huge list of error messages and where you can find documentation on how to deal with them.  
- 1994 version of the NOS 2 operations handbook
- NOS 2 Reference manual volume 2 which is a good overview on NOS usaage 
- NOS 2 Reference manual volume 1 which covers use of IAF.
-  the 1992 version of the NOS 2 Reference manual volume 4, system interface. 
- CDC Seminar on learning CPU COMPASS programming
- COMPASS instant
- most recent COMPASS reference manual
- COMPASS 3.0 internal maintenance specification
-  Cyber Loader reference manual
- Common memory manager reference manual
- 3 manuals on the Cyber Record Manager


I also added manuals for UPDATE and MODIFY Since source management tools are frequently used. 

you can find them all in one place at https://drive.google.com/drive/folders/1Wn3_kJwr1hFodZlUn12Ivta4_6POB17w?usp=drive_link

**Reply by KC5VDJ** [reply]
Thanks!  When I do something, I don't do half-assed.. 😀

I do believe I was missing 60459300Y, what sucks is how far I am into revision U. 🤬

I would still like to find UT-Lisp's though...  The 1975 version is as good as I can find.  I know where one used to be, matching the version in the 2.8.7 build, but the whole "range instrumentation systems department" thing means anything would have probably been pulverized, due to site classification (S).

It's in a box somewhere.  Black plastic edge binding, the thin, pressed, melted tight kind?  The place that box is in looks kinda small from 50,000 nmi.  Oh well... Error while uploading Artemis.1-Earth.at.50k.miles-landscape.webp

**Reply by KC5VDJ** [reply]
I forgot to add, that if anyone wants to contribute, a decision for later, I put a bulletin into the main with some simple ground rules for potential contributors.  Of course that itself is pdf.

**Comment by KC5VDJ** [comment]
Be looking for 2.8.7 "System Commands" soon... It took precedence 😉

Up to section 4 now...

**Reply by KC5VDJ** [reply]
> Be looking for 2.8.7 "System Commands" soon... It took precedence 😉
> 
> Up to section 4 now...

Ummm...  Does anyone other than CDC consider the PROC COMMENT (.*)  Directive to be "Iterative"?

Ummm, I am correcting the obvious mistakes in nesting.

**Comment by KC5VDJ** [comment]
Is there one explaining the default USECPU,0 scheduling in a -2 model machine?  SMP? Sorta-SMP? Braindead fixed alternation?

Options 1 and 2 are clearly explicitly affine as specified, quite useful as well (visions of a bastardized NOS/GUARDIAN), but my question is about the default "Use either central processor" condition, and how well that would co-exist with badly written explicitly-affine systems (GTel, I won't name it, but one of the developers whom I was later a team member with told them it wouldn't work, VMS to a Guardian cluster-f (square pegs / round holes))....digressing again...  In a nutshell the flow for an incoming job on how a CPU is automatically selected for a waiting job to execute in through the control being handed to that execution.

Conceptualizing, also, I keep drawing some parallels to the 11/782...is it just me, or did DEC take liberties?

> I've just taken the time to look at your version of the installation guide manual. This is great work!  
> 
> I've put a few manuals on my google drive that I think are the most essential to have as searchable manuals for day to day usage/administration of NOS2.8.7
> 
> You will find:
> 
> -  the most recent system programmer's instant
> -  the 1997 edition of volume 3 (system commands)
> - the 1996 Analysis handbook. 
> -  1988 version of the system programmer's instant
> - the diagnostic index. which contains a huge list of error messages and where you can find documentation on how to deal with them.  
> - 1994 version of the NOS 2 operations handbook
> - NOS 2 Reference manual volume 2 which is a good overview on NOS usaage 
> - NOS 2 Reference manual volume 1 which covers use of IAF.
> -  the 1992 version of the NOS 2 Reference manual volume 4, system interface. 
> - CDC Seminar on learning CPU COMPASS programming
> - COMPASS instant
> - most recent COMPASS reference manual
> - COMPASS 3.0 internal maintenance specification
> -  Cyber Loader reference manual
> - Common memory manager reference manual
> - 3 manuals on the Cyber Record Manager
> 
> 
> I also added manuals for UPDATE and MODIFY Since source management tools are frequently used. 
> 
> you can find them all in one place at https://drive.google.com/drive/folders/1Wn3_kJwr1hFodZlUn12Ivta4_6POB17w?usp=drive_link

**Reply by wschaub** [reply]
The reason for USECPU is for systems like the 6700 where the second CPU had different capabilities. I think for example on some models with dual CPUs only CPU1 will have the compare move unit.  on a 6700 the second CPU is a 6400 type serial CPU vs the main 6600 CPU with the scoreboard, etc. 

USECPU,0. just means any CPU and the scheduling is going to look at a lot like it's described in the NOS 1.4 IMS 3 volume set. 

https://bitsavers.org/pdf/cdc/cyber/nos/60454300B_NOS_1.4_Internal_Maintenance_Spec_Vol_1_Aug79.pdf
https://bitsavers.org/pdf/cdc/cyber/nos/60454300B_NOS_1.4_Internal_Maintenance_Spec_Vol_2_Aug79.pdf
https://bitsavers.org/pdf/cdc/cyber/nos/60454300B_NOS_1.4_Internal_Maintenance_Spec_Vol_3_Aug79.pdf

The NOS Analysis course also gives a lot of good background to the above. 
https://bitsavers.org/pdf/cdc/cyber/nos/FH4010-1C_NOSanalysis_Apr80.pdf
https://bitsavers.org/pdf/cdc/cyber/nos/FH4010-4C_StudyDump_Apr80.pdf

The KRONOS 2.1 workshop is also a useful read since NOS1 basically started life as KRONOS renamed. 
https://bitsavers.org/pdf/cdc/cyber/cyber_70/kronos/97404700D_KRONOS2.1wrksp76.pdf

There are differences between NOS1 and NOS2 like negative field length for tracking things that used to be in the global FNT, EST tables having an extra  word per entry now, etc.  but the IMS is still a useful tool to learn NOS internals and a surprising amount of it still lines up with NOS2.
